packages/js/product-editor/src/components/header/plugin-header-items/plugin-header-item-modal.tsx

```diff
@@ -0,0 +1,44 @@
+/**
+ * External dependencies
+ */
+import { createElement, Fragment, useState } from '@wordpress/element';
+import { Button, Modal } from '@wordpress/components';
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore No types for this exist yet.
+// eslint-disable-next-line @woocommerce/dependency-group
+import { PinnedItems } from '@wordpress/interface';
+
+/**
+ * Internal dependencies
+ */
+import { PINNED_ITEMS_SCOPE } from '../../../constants';
+import { PluginHeaderItemModalProps } from './types';
+
+export const PluginHeaderItemModal: React.FC< PluginHeaderItemModalProps > = ( {
+	children,
+	label,
+	icon,
+	title,
+} ) => {
+	const [ isOpen, setOpen ] = useState( false );
+	return (
+		<PinnedItems scope={ PINNED_ITEMS_SCOPE }>
+			<>
+				<Button
+					variant="tertiary"
+					icon={ icon }
+					label={ label }
+					onClick={ () => setOpen( ! isOpen ) }
+				/>
+				{ isOpen && (
+					<Modal
+						title={ title }
+						onRequestClose={ () => setOpen( false ) }
+					>
+						{ children }
```

Comment:
We should pass a callback to the children to allow the extension to close the modal (for example, after a button is clicked on the modal).

Reply:
Done in commit [3969d4a](https://github.com/woocommerce/woocommerce/pull/43687/commits/3969d4ab9dae82f9ebe574171e3e27fb8e074192)